pilot/pkg/model/sidecar.go

```diff
@@ -47,6 +47,11 @@ var (
 		kind.VirtualService,
 		kind.DestinationRule,
 		kind.Sidecar,
+
```

Comment:
we never call AddConfigDependencies for any of these, so I think this is broken and will skip recomputation always.

Reply:
We do record gateway-api dependencies
https://github.com/istio/istio/blob/3cc06285111ff3a69e72a1fdda77eb8825b54e62/pilot/pkg/model/virtualservice.go#L550

Reply:
oh nice, thanks!

A test would probably be helpful but lgtm